python/docs/source/user_guide/sql/python_udtf.rst

```diff
@@ -0,0 +1,222 @@
+            ...
+
+
+The return type of the UDTF defines the schema of the table it outputs. 
+It must be either a ``StructType`` or a DDL string representing a struct type.
+
+**Example UDTF Implementation:**
+
+Here is a simple example of a UDTF implementation:
+
+.. literalinclude:: ../../../../../examples/src/main/python/sql/udtf.py
+    :language: python
+    :lines: 38-40
+    :dedent: 4
+
+
+**Instantiating the UDTF:**
+
+To make use of the UDTF, you'll first need to instantiate it using the `udtf` function:
+
+.. literalinclude:: ../../../../../examples/src/main/python/sql/udtf.py
+    :language: python
+    :lines: 42-51
+    :dedent: 4
+
+In this example, `SimpleUDTF` is instantiated with the return columns `c1` and `c2`, both of type `int`. 
+
+**Using `udtf` as a Decorator:**
+
+An alternative way for implementing a UDTF is to use `udtf` as a decorator:
+
+.. literalinclude:: ../../../../../examples/src/main/python/sql/udtf.py
+    :language: python
+    :lines: 56-68
+    :dedent: 4
+
+For more detailed usage, please see :func:`udtf`.
+
+
+Registering and Using Python UDTFs in SQL
+-----------------------------------------
+
+Python UDTFs can also be registered and used in SQL queries.
+
+.. literalinclude:: ../../../../../examples/src/main/python/sql/udtf.py
+    :language: python
+    :lines: 72-98
+    :dedent: 4
+
+
+Arrow Optimization
+------------------
+Apache Arrow is an in-memory columnar data format used in Spark to efficiently transfer
+data between JVM and Python processes. Apache Arrow is disabled by default for Python UDTFs.
+
+To enable Arrow optimization, set ``spark.sql.execution.pythonUDTF.arrow.enabled`` to ``true``.
+Alternatively, you can specify the `useArrow` parameter when declaring the UDTF:
+
+.. literalinclude:: ../../../../../examples/src/main/python/sql/udtf.py
+    :language: python
+    :lines: 103-108
+    :dedent: 4
+
+
+For more details, please see `Apache Arrow in PySpark <../arrow_pandas.rst>`_.
+
+
+More Examples
+-------------
+
+A Python UDTF with `__init__` and `terminate`:
+
+.. literalinclude:: ../../../../../examples/src/main/python/sql/udtf.py
+    :language: python
+    :lines: 113-132
+    :dedent: 4
+
+
+A Python UDTF to generate a list of dates:
+
+.. literalinclude:: ../../../../../examples/src/main/python/sql/udtf.py
+    :language: python
+    :lines: 137-173
+    :dedent: 4
+
+
+Advanced Features
+-----------------
+
+TABLE input argument
+~~~~~~~~~~~~~~~~~~~~
+Python UDTFs can also take a TABLE as input argument, and it can be used in conjunction 
+with scalar input arguments.
+By default, you are allowed to have only one TABLE argument as input, primarily for 
+performance reasons. If you need to have more than one TABLE input argument, 
+you can enable this by setting ``spark.sql.tvf.allowMultipleTableArguments.enabled`` to ``true``.
+
+.. literalinclude:: ../../../../../examples/src/main/python/sql/udtf.py
+    :language: python
+    :lines: 178-197
+    :dedent: 4
```

Comment:
@ueshin @dtenedor I've added this TABLE argument support under the advanced feature section. We should add more examples here and in the udtf function docstring.

Reply:
Created [SPARK-44746](https://issues.apache.org/jira/browse/SPARK-44746)